code/__DEFINES/atmospherics/atmos_piping.dm

```diff
@@ -97,3 +97,20 @@
 #define VENTCRAWL_ENTRANCE_ALLOWED (1<<1)
 ///Used to check if a machinery is visible. Called by update_pipe_vision(). On by default for all except cryo.
 #define VENTCRAWL_CAN_SEE	(1<<2)
+
+/**
+ * During mapload and before smoothing happens, this will add src to the blueprint data (or queue, to be added during icon smoothing) if necessary.
+ * Args:
+ * loc - The location of the turf to add the blueprints to.
+ */
+#define ADD_BLUEPRINTS_PREROUND(loc) \
+	if(mapload && !SSicon_smooth.initialized) { \
+		var/turf/_turf_loc = loc; \
+		if(isturf(_turf_loc)) { \
+			if(layer == WIRE_LAYER) { \
+				SSicon_smooth.blueprint_queue += src \
+			} else { \
+				_turf_loc.add_blueprints(src) \
+			}; \
+		}; \
+	}; \
```

Comment:
I really don't think this justifies being a macro, just makes it harder to read and change for light optimization, alongside removing our ability to track call count and such

Reply:
I had checked the profiler before and there was close to like 30 calls on this on Sybil, but each time you placed a pipe it called this which I thought was really unnecessary. I wanted to add the mapload check but my options are to either have mapload as an arg in ``add_blueprint_preround`` which I think would suck or it would be to add a check to every individual call of it which I thought was just more copy paste I didn't want to contribute to

Reply:
so does this change just exist to avoid a proc call on initialize then? it's not like the call does anything right now cause SSicon_smooth is init'd by that point, right?

Reply:
Yeah that was the premise of it, I just thought it was weird we're calling this proc every time a pipe is manually created in a round just to do absolutely nothing

Reply:
I'm sorry then but this is silly. the mapload doesn't need to be there cause if your concern is the proc call then we can just make the proc into a macro, but that's also kinda silly cause like, the proc call doesn't matter, right?

Reply:
I'll try doing that when it's time for it